src/Prelude/Linear.hs

```diff
@@ -74,3 +83,29 @@ maybe _ f (Just y) = f y
 -- arrow but we have a linear arrow
 forget :: (a ->. b) ->. a -> b
 forget f x = f x
+
+-- | Linearly typed replacement for the standard 'foldr' function, to allow
+-- linear consumption of a list.
+foldr :: (a ->. b ->. b) -> b ->. [a] ->. b
+foldr f z = \case
+  [] -> z
+  x:xs -> f x (foldr f z xs)
+
+-- | Replacement for the flip function with generalized multiplicities.
+flip :: (a -->.(p) b -->.(q) c) -->.(r) b -->.(q) a -->.(p) c
+flip f b a = f a b
+
+-- | Linearly typed replacement for the standard '(<$)' function.
+(<$) :: (Applicative f, Consumable b) => a ->. f b ->. f a
+a <$ fb = pure (flip lseq a) <*> fb
+
+-- | Linearly typed replacement for the standard '(<*)' function.
+(<*) :: (Applicative f, Consumable b) => f a ->. f b ->. f a
+fa <* fb = fmap (flip lseq) fa <*> fb
```

Comment:
I wanted to place `(<*)` and `(<$)` in `Control.Monad` but that creates a circular dependency because of the use of Consumable.

Reply:
Time for more crazy `Internal` cycle breaking? Or maybe these have their rightful place in the prelude?

If cycle breaking is preferable: don't hold this PR, and make an issue for it.

Reply:
I think `Internal` is a sensible thing to do here, so that they could be imported qualified by `Control` when the `Data` version is also necessary

Reply:
On further thoughts (apologies I missed this earlier), both `(<*)` and `(<$)` can be given more general type signatures:
```haskell
(<$) :: (Control.Functor f, Consumable b) => a ->. f b ->. f a
a <$ fb = Control.fmap (`lseq` a) fb

(<*) :: (Data.Applicative f, Consumable b) => f a ->. f b ->. f a
fa <* fb = Data.fmap (flip lseq) fa Data.<*> fb
```
(and these cover the ones you've given).